tempesta_fw/addr.c

```diff
@@ -205,87 +221,24 @@ tfw_inet_pton_ipv6(char **p, struct sockaddr_in6 *addr)
 /**
  * Parse IPv4 and IPv6 addresses with optional port.
  * See RFC5952.
- *
- * @p - string pointer, updated by the function.
- * @addr - distination to write as a pointer to a union of sockaddr_in and
- * 	   sockaddr_in6.
  */
 int
-tfw_inet_pton(char **p, void *addr)
+tfw_inet_pton(const char *str, TfwAddr *addr)
 {
-	int mode = 0;
-
-	/* Eat empty string prefix. */
-	while (**p && isspace(**p))
-		++*p;
+	memset(addr, 0, sizeof(*addr));
 
-	/* Determine type of the address (IPv4/IPv6). */
-	if (**p == '[' || isalpha(**p)) {
-		mode = 6;
-	} else {
-		char *p1 = *p;
-		while (*p1 && isdigit(*p1))
-			p1++;
-		if (*p1 == ':') {
-			mode = 6;
-		}
-		else if (*p1 == '.') {
-			mode = 4;
-		}
-		else {
-			TFW_ERR("bad string: %s\n", *p);
-			return -EINVAL;
-		}
-	}
+	/* The IPv6 address must be enclosed into square brackets,
+	 * or else we can't distinguish it from the port. */
+	if (str[0] == '[' && !strcspn(str, "1234567890ABCDEFabcdef:[]"))
+		return tfw_inet_pton_ipv6(str, &addr->v6);
 
-	if (mode == 4)
-		return tfw_inet_pton_ipv4(p, addr);
-	if (mode == 6)
-		return tfw_inet_pton_ipv6(p, addr);
+	if (!strcspn(str, "1234567890.:"))
+		return tfw_inet_pton_ipv4(str, &addr->v4);
 
-	TFW_ERR("Can't parse address %s\n", *p);
 	return -EINVAL;
 }
-/* TODO: Remove the export after implementing a configuration framework
- * that will remove the dependency of tfw_sched_http on this function. */
 EXPORT_SYMBOL(tfw_inet_pton);
 
-/**
- * Print UPv4/IPv6 address in network byte order to @buf.
- * @buf must be MAX_ADDR_LEN bytes in size.
- */
-int
-tfw_inet_ntop(const void *addr, char *buf)
```

Comment:
Now we have a faster `tfw_addr_fmt()` that does the same thing but without `snprintf()`.
The functions were different, but at some point they started doing the same thing.
In this patch I removed the last reference to the old `tfw_inet_ntop()` so I removed the function.

Also, there is a little naming inconsistency here: `tfw_inet_*()` vs `tfw_addr_*()`.
For me `tfw_addr_*()` looks more suitable for the `addr.c` and `TfwAddr`.
Should I re-name the old `tfw_inet_` functions?